src/core/src/index/revindex/mod.rs

```diff
@@ -425,20 +425,20 @@ impl Datasets {
         }
     }
 
-    fn from_slice(slice: &[u8]) -> Option<Self> {
+    pub fn from_slice(slice: &[u8]) -> Result<Self> {
         use byteorder::ReadBytesExt;
 
         if slice.len() == 8 {
             // Unique
-            Some(Self::Unique(
+            Ok(Self::Unique(
                 (&slice[..]).read_u32::<LittleEndian>().unwrap(),
```

Comment:
Can we take out the unwrap here?

```suggestion
            Ok(Self::Unique((&slice[..]).read_u32::<LittleEndian>()?))
```

maybe not without adding a new error type in `sourmash::Error`, so OK for now but might want to review in the future

Reply:
will create an issue, thx.